waiter/src/waiter/state/maintainer.clj

```diff
@@ -583,18 +584,19 @@
                                                  (count services-without-instances) "services without instances:"
                                                  (vec services-without-instances)))
                                      (ejection-expiry/select-services! ejection-expiry-tracker all-available-service-ids')
-                                     (assoc loop-state
-                                       :all-available-service-ids all-available-service-ids'
-                                       :service-id->healthy-instances service-id->healthy-instances'
-                                       :service-id->unhealthy-instances service-id->unhealthy-instances'
-                                       :service-id->expired-instances service-id->expired-instances'
-                                       :service-id->starting-instances service-id->starting-instances'
-                                       :service-id->failed-instances service-id->failed-instances'
-                                       :service-id->instance-counts service-id->instance-counts'
-                                       :service-id->deployment-error service-id->deployment-error'
-                                       :service-id->instability-issue service-id->instability-issue'
-                                       :service-id->killed-instances service-id->killed-instances'
-                                       :time scheduler-sync-time))
+                                     [(assoc loop-state
```

Comment:
We already have `service-id->failed-instances`, I do not think this needs to change.